### PR TITLE
Support tf v.12, tunnel-only users, egress limiting, static ssh users…

### PR DESCRIPTION
**Pull request by eyablonowitz**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -243,9 +240,7 @@ resource "aws_autoscaling_group" "bastion_auto_scaling_group" {
   min_size             = "${var.bastion_instance_count}"
   desired_capacity     = "${var.bastion_instance_count}"
 
-  vpc_zone_identifier = [
-    "${var.auto_scaling_group_subnets}",
-  ]
+  vpc_zone_identifier = "${var.auto_scaling_group_subnets}"
```
Comment (efredrickson-patientping) [inline main.tf]: Nitpick: Variable references. I see some variables you switched to or added are using the TF 0.12 syntax, a handful like this are still using the interpolation syntax when just being referenced.

File: user_data.sh [hunk 2]
```diff
@@ -16,6 +16,19 @@ setfacl -Rdm other:0 /var/log/bastion
 # Make OpenSSH execute a custom script on logins
 echo -e "\\nForceCommand /usr/bin/bastion/shell" >> /etc/ssh/sshd_config
 
+sed -i 's/MaxAuthTries\ [0-9]/MaxAuthTries 5/' /etc/ssh/sshd_config
+
+# Deny interactive shell to some users (tunnel-only)
+cat >> /etc/ssh/sshd_config << 'EOF'
```
Comment (efredrickson-patientping) [inline user_data.sh:22]: This all looks sane. Though we do have some compliance stuff that will need to be addressed and some existing configs/opinions regarding the sshd config (in the sysadmin repo). These are things like the login banner and session timeout.

File: main.tf [hunk 3]
```diff
@@ -229,7 +221,12 @@ resource "aws_launch_configuration" "bastion_launch_configuration" {
     "${aws_security_group.bastion_host_security_group.id}",
   ]
 
-  user_data = "${data.template_file.user_data.rendered}"
+  user_data = templatefile("${path.module}/user_data.sh", {
```
Comment (efredrickson-patientping) [inline main.tf:224]: If we set up a bastion stack with a pre-built or ansible-managed AMI we should consider disabling injecting user-data.

**Review by efredrickson-patientping — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (eyablonowitz) [reply main.tf]: Fixed

File: user_data.sh (hunk 2 above)
Comment (eyablonowitz) [reply user_data.sh:22]: Not an issue for my current non-interactive use case, but I'll address in another PR if we use this module to create human-access bastion.

File: main.tf (hunk 3 above)
Comment (eyablonowitz) [reply main.tf:224]: Makes sense.  Can address in another PR if we choose to go down that route.

**Pull request merged**